plugins/containers-plugin/package.json

```diff
@@ -33,22 +33,26 @@
     "@eclipse-che/plugin": "latest",
     "rimraf": "2.6.2",
     "typescript-formatter": "7.2.2",
-    "typescript": "2.9.2"
+    "typescript": "2.9.2",
+    "ts-loader": "^4.1.0",
+    "clean-webpack-plugin": "^0.1.19",
+    "webpack": "^4.1.1",
+    "webpack-cli": "^3.1.1"
   },
   "dependencies": {
   },
   "scripts": {
     "prepare": "yarn run clean && yarn run build",
     "clean": "rimraf lib",
     "format-code": "tsfmt -r",
-    "watch": "tsc -watch",
-    "compile": "tsc",
+    "watch": "webpack-cli -w --config webpack.config.js",
+    "compile": "webpack-cli --config webpack.config.js",
     "build": "yarn run format-code && yarn run compile && theia:plugin pack"
   },
   "engines": {
     "theiaPlugin": "next"
   },
   "theiaPlugin": {
-    "backend": "lib/containers-plugin.js"
+    "frontend": "dist/containers-plugin.js"
```

Comment:
why move it to be a front-end plug-in ?
it will make more stuff to load on client side for almost nothing useful for this plug-in ?

Reply:
@benoitf as for me, client side plugin is more reasonable. At least it reduces client-server traffic.

Reply:
did you analyze traffic ?
It makes client download more stuff

Reply:
@benoitf  Totally agree with you about optimally using it as a server-side plugin. So I did it as a server-side plugin in the first implementation.

Some explanations: during the last sprint, we have a lot of problem with 'eclipse/che-theia' image with '@eclipse-che/theia-factory-plugin', because our the API depends on the version of 'ws-skeleton/che-theia-generator' (https://github.com/ws-skeleton/che-theia-generator/blob/master/src/conf/extensions.yml) which adds implementations of our CHE API... 
We had a problem, but couldn't catch them before starting using an editor.
    And I'm convinced about the necessity to minimal covering by che-selenium tests for theia-ide plugins ASAP because we still don't have a QA strategy for testing our theia-plugins which are using CHE API.
    For this moment, our theia-ide editor has two backend plugins which are using CHE API: theia-containers-plugin and theia-factory-plugin.  And it will be nice to cower them by che-selenium tests.  But in the case with one frontend plugin, it covers both:  frontend and backend.
That is why I rework my containers plugin to the frontend plugin.

WDYT?

Reply:
hello @olexii4 
AFAIK production plug-in are not tests so they should behave in the most efficient manner for end users.
BTW che-theia-samples has examples that could be used for tests as well

Reply:
ok

Reply:
@vitaliy-guliy if there is no need to run on client side (the callbacks) then it should be on server side to avoid to download extra .js when loading the page